src/scan/postgres_scan.cpp

```diff
@@ -170,10 +182,15 @@ PostgresScanGlobalState::PostgresScanGlobalState(Snapshot _snapshot, Relation _r
 	ConstructTableScanQuery(input);
 	table_reader_global_state =
 	    duckdb::make_shared_ptr<PostgresTableReader>(scan_query.str().c_str(), count_tuples_only);
+	duckdb_pg_scan_memory_ctx =
+	    AllocSetContextCreate(CurrentMemoryContext, "DuckDBPerTupleContext", ALLOCSET_DEFAULT_MINSIZE,
+	                          ALLOCSET_DEFAULT_INITSIZE, ALLOCSET_DEFAULT_MAXSIZE);
```

Comment:
This is incorrect. Every postgres scan will now create a separate memory context, which they all assign to the same global variable. The cleanup to set to nullptr is also dangerous, if one scan finishes before another. I think what would be better is to do the following:
1. Storing this context actually in the "global state" of the scan, instead of storing it into a process global variable.
2. Switch to this context right after we take the `GlobalProcessLock` in `PostgresScanFunction`, and switching back right before we release it.

Also the name of the context could be improved imo
```suggestion
	duckdb_pg_scan_memory_ctx =
	    AllocSetContextCreate(CurrentMemoryContext, "DuckdbScanContext", ALLOCSET_DEFAULT_MINSIZE,
	                          ALLOCSET_DEFAULT_INITSIZE, ALLOCSET_DEFAULT_MAXSIZE);
```

Reply:
ah yes, you're right.